### ISSUE TITLE
POST API calls don't update sheet if it is not open in a browser tab (with threads version)

### ISSUE THREAD

**Issue by balajeerc** [audreyt/ethercalc#439]
Steps to reproduce:
1. Create a sheet. Open it in browser.
2. Enter a series of random numbers in the first column. (i.e A)
3. Close the browser session.
4. Make the following POST call:
   `
   { 
       "command": ["sort A1:A10 A up"]
   }
   `
5. Open the sheet in the tab again. Note that the sort has not been applied.

If you skip step 3 in the above sequence, then the sort call applies correctly.

It seems that the API calls fail to update without a client session being open.

**Comment by eddyparkinson** [comment]
It works for me.

curl -H "Content-Type: application/json" -X POST  http://sheet.cellmaster.com.au/_/status -d '{"command":"sort B1:B4 B up"}'

I did as you suggested and closed the tab.
I also tried with a tab that has never been opened since a server restart, also worked.

I sometimes add '{ "room": " status", "snapshot": ""}', this creates the room if it does not exist.

Also, I have code that updates a sheet when the browser window is closed. Always looks to work, never seen a problem.

**Comment by balajeerc** [comment]
@eddyparkinson 

Thanks for taking a look at this.


I played around with this a little more. Is it possible you are running with webworker-threads disabled when you ran the tests? When I run with the `--vm` option, everything seems to work fine.
When I run with webworker-threads enabled, it seems to start manifesting the problems described above.

If not, I suppose it's something local to my installation.

The ethercalc version I am using is whatever is installed from the npm repos.
Peering into `package.json`, I see the version listed as `"version": "0.20160612.0"`.
The version of node I am using: 6.2.1
The version of redis I am using: 2.8.4
Ubuntu 14.04, x86_64

I won't re-open this. Am just leaving the information here in case someone runs into this problem.

**Comment by eddyparkinson** [comment]
yes, the version I tested against is using node 0.1 - so no threads if I remember. (runs on Openshift)

**Comment by eddyparkinson** [comment]
webworker-threads - balajeerc  says it fails then webworker-threads are turned on. Reopening issue.
